### PR TITLE
Add method to replace data for an already rendered chart

### PR DESCRIPTION
**Pull request by jeffandersen**
I needed a method exposed so that the data could be updated on the fly, not by drag and drop

And while I'm not sure it's correct, I've included a fix for what I believe was a bug. The Y value when defined should be used, otherwise, normalize (calculate) the value from X.

### CODE REVIEW

**Comment by jeffandersen** [comment]
The demo can be tested by running `npm run build` and changing the dist script include to point to `../dist` instead of the CDN

But here's a video:
https://www.loom.com/embed/582073a254034a968fa4af12cdc5c499

**Comment by nagi1** [comment]
Please elaborate more on your idea

File: __test__/index.test.js [hunk 1]
```diff
@@ -38,27 +38,53 @@ beforeEach(() => {
       description: 'Gettin there',
       size: 10,
       x: 55.11627906976744,
-      y: 44.88372093023257,
```
Comment (nagi1) [inline __test__/index.test.js]: At least keep one Y defined to pass the test below "uses the supplied y value if defined"

File: __test__/index.test.js [hunk 2]
```diff
@@ -155,3 +181,26 @@ describe('hillchart@render', () => {
     expect(svg.getElementsByClassName('hill-chart-text')[1]).toBeUndefined();
   });
 });
+
+describe('hillchart@replaceData', () => {
```
Comment (nagi1) [inline __test__/index.test.js:186]: Have you replaced `describe('hillchart@render')`?

File: src/index.js [hunk 3]
```diff
@@ -204,23 +245,6 @@ export default class HillChart extends EventEmitter {
       .attr('y', calculateTextMarginForY());
   }
```
Comment (nagi1) [inline src/index.js:247]: What changed here?

**Review by nagi1 — CHANGES_REQUESTED** [review]
Hey, thanks for taking the time to submit this.
I highlighted somethings to be reviewed and do you have any idea why the test covrage in coveralls droped?

File: __test__/index.test.js (hunk 2 above)
Comment (jeffandersen) [reply __test__/index.test.js:186]: No, that still exists at line 120

File: src/index.js (hunk 3 above)
Comment (jeffandersen) [reply src/index.js:247]: I ordered the "Render" functions at the same point in the file, `undraggablePoint` got moved up the file

Comment (jeffandersen) [reply src/index.js:247]: You can see it up at line 103 now

File: __test__/index.test.js (hunk 1 above)
Comment (jeffandersen) [reply __test__/index.test.js]: I've now bought this point back

**Comment by jeffandersen** [comment]
I think the coverage had decreased because I had removed the testing of `y` from Render... now that there are some tests for that, it has come back up to par

**Review by nagi1 — COMMENTED** [review]
cool, now user doesnt have to enter the value of Y at all?

**Comment by jeffandersen** [comment]
Right, it will calculate it based on the X value if there is no value supplied

**Comment by nagi1** [comment]
Cool thanks for your time, if you liked this project and have some spare time look at the to-do list as well as leaving a feedback that would be helpful :)

**Comment by jeffandersen** [comment]
Thanks, I have some ideas for additional features as well.. will consider those when I have some spare time.

@nagi1 do you have a sense of when a new version will make it to npm?

**Comment by nagi1** [comment]
I dont like that nasty `drag()` function, which by the way spent a lot of time trying to rewrite it without any luck.
Maybe after fixing it we can move to some more exiting ideas.

As of the npm version it'll be available right away.

Happy coding :)

**Pull request merged**